### PR TITLE
Understanding Gazpacho in Python and its Application in Web Scraping

### LINKED ISSUE #6269: [Languages]Understanding Gazpacho in Python and its Application in Web Scraping
#### Proposed title of article
Understanding Gazpacho in Python and its Application in Web Scraping
#### Proposed article introduction
Web Scraping is considered a fundamental process of getting data from the web. It automates the process of extracting the data from a web page, which is quicker and hassle-free than the conventional copy-pasting of the data. 
In Python, Web Scraping is done predominantly using two libraries, requests and BeautifulSoup. Web scraping using gazpacho can come really handy when we want to fetch data in no time. This article will cover as much concerning  the gazpacho library and  its implementation in  scraping  web data .

#### Key takeaways 
The reader will be able to:
- Get the understanding of the gazpacho library and how it works.
- Understand web scraping and its relevance.
- Understand web scraping in Python.
- Understand how gazpacho library is used in web scraping.

#### Article quality
This article will cover as much to help the reader not only understand web scraping and its relevance but also how we can use the gazpacho library in python in web scraping.

#### References
N/A

### Templates to use as guides
- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

### PR DESCRIPTION
**Pull request by dentex22**
## Our **Peer Reviewer and Content Moderator teams do NOT provide any revisions services.** All revisions and edits should be completed on your own forked repo (as to not take up room in the queue) in order for our team to review them all in a timely manner. 

**ANY ARTICLE SUBMITTED WITH GLARING ERRORS WILL BE IMMEDIATELY CLOSED.**

>As a rule of thumb - please be sure to only submit articles (pull requests) that are fully polished and ready to be published. Be sure to go through our resources documents for extra and 3rd party (vetted) resources to help improve overall technical writing.

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is **600x400 pixels**, under **300KB in size**, and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

This closes #6269

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

**Comment by srishilesh** [comment]
Hello @dentex22 ,

Thanks for submitting this PR.

Your PR has entered our reviewing queue. Please wait until a peer-reviewer pings you to get started.

Meanwhile, we suggest you to go through existing approved PRs/published articles to see if you have missed out anything.

Thanks.

**Comment by dentex22** [comment]
> Hello @dentex22 ,
> 
> Thanks for submitting this PR.
> 
> Your PR has entered our reviewing queue. Please wait until a peer-reviewer pings you to get started.
> 
> Meanwhile, we suggest you to go through existing approved PRs/published articles to see if you have missed out anything.
> 
> Thanks.

ook

**Comment by louisefindlay23** [comment]
Plagiarism check complete

**Comment by dentex22** [comment]
> @dentex22 Please perform a grammarly check on your tutorial. I've found lots of grammatical errors.

ook

**Comment by dentex22** [comment]
> @dentex22 Please perform a grammarly check on your tutorial. I've found lots of grammatical errors.

done with changes

**Comment by dentex22** [comment]
> > > @dentex22 Please perform a grammarly check on your tutorial. I've found lots of grammatical errors.
> > 
> > 
> > done with changes
> 
> @dentex22 Have you used [Grammarly](https://app.grammarly.com/) to help you perform the check? I can still see the same grammatical mistakes.

hello @Bayler  i used the [Hemingway](https://hemingwayapp.com/), to fix the grammatical errors , but have now fixed using the Grammarly as instructed, you can take a look at it.

**Comment by Bayler** [comment]
plag

I understand the code bit. But, what about these sentences highlighted in red? Let's not waste a lot of time here. It steals valuable time I'd use reviewing polished PRs.

File: content/articles/understanding-gazpacho-in-python-and-its-application-in-webscraping/index.md [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+- Selenium is a collection of programs rather than a single tool. It's a free testing tool for web apps that works across several browsers and platforms.
+- Scrapy is a fast, open-source web crawling system that uses selectors based on XPath to extract data from online pages.
+- Gazpacho is a modern, simple, and fast web scraping library. On this [site](https://www.analyticsvidhya.com/blog/2020/04/5-popular-python-libraries-web-scraping/), you can get to know more about the same.
+
+### Gazpacho library
+In this tutorial, we will focus on how we can use the Gazpacho library to automate data extraction from web pages and websites. This is a modern web scraping Python library, stable and installed with zero dependencies. It can combine the functionality of the requests and Beautiful Soup libraries by importing a few classes from each.
+
+To use this library, check if Python is installed on your machine by typing the command `"python"` in the command prompt. If it is not installed, go to this [page](https://www.python.org/downloads/.) to download and install it.
+
+Once Python is installed and running, we can then install the library using the pip command below.
```
Comment (Bayler) [inline content/articles/understanding-gazpacho-in-python-and-its-application-in-webscraping/index.md:79]:
```suggestion
Once Python is installed and running, we can install the library using the `pip` command below:
```

File: content/articles/understanding-gazpacho-in-python-and-its-application-in-webscraping/index.md [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+- Content aggregators, such as news aggregators and employment aggregators, frequently use web scraping to provide updated data to their subscribers.
+- Collecting data for sales and marketing purposes, such as emails and phone numbers.
+- Used in SEO tools for website optimization for search engines.
+- Data retrieval for machine learning projects.
+
+### Web Scraping in Python
+Web scraping involves a wide variety of programming technologies and techniques. In Python, web scraping is done with various libraries, that is:
+- Requests: It is the most basic and essential library for web scraping, used for making various types of HTTP requests like GET and POST. However, with requests, it does not parse the HTML data retrieved, so libraries like Beautiful Soup are required.
+- lxml: Since the request library cannot parse HTML retrieved from webpages, lxml is required. A fast, production-quality, and high-performance HTML and XML parsing library
+- Beautiful Soup is the most extensively used scraping library and produces parse trees for reading HTML and XML data and converts incoming and outgoing documents to Unicode and UTF-8 automatically.
```
Comment (Bayler) [inline content/articles/understanding-gazpacho-in-python-and-its-application-in-webscraping/index.md:69]: Such long sentences can be hard to follow. Also break such long sentences into two.

File: content/articles/understanding-gazpacho-in-python-and-its-application-in-webscraping/index.md [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+- 
+#### Operation of the Web Scraper
+For the web scraper to perform and accomplish the scraping process, it goes through various steps as stated below:
+- First, the web scraper downloads the requested content from multiple websites and web pages through the use of a web crawler.
+- Since most of the data from websites are HTML and unstructured, the web scraper will parse the downloaded content and extract structured data.
+- When data is extracted from an unstructured form into a structured form, the scraper saves it in one of several formats, such as CSV, JSON, or a database.
+- When the structured data has been stored, the web scraper will analyze the data obtained.
+
+#### Applications for Web Scraping
+Web scraping is utilized in a variety of real-world scenarios, including:
```
Comment (Bayler) [inline content/articles/understanding-gazpacho-in-python-and-its-application-in-webscraping/index.md:58]: Always use active tenses. For example, you could begin this sentence with...."We utilize web scraping.....". Go through the whole article and correct accordingly.

File: content/articles/understanding-gazpacho-in-python-and-its-application-in-webscraping/index.md [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+- Data transformation and cleaning module: The data collected by the extractor isn't ready to use right away; it needs to go through some sort of cleaning process before we can utilize it. For this, methods such as string manipulation and regular expressions can be employed. Extraction and transformation can be done in one process.
+- Once we have extracted the needed data, it is stored in standard formats that can be stored in the database, in JSON, or CSV format in the storage module.
+- 
+#### Operation of the Web Scraper
+For the web scraper to perform and accomplish the scraping process, it goes through various steps as stated below:
+- First, the web scraper downloads the requested content from multiple websites and web pages through the use of a web crawler.
+- Since most of the data from websites are HTML and unstructured, the web scraper will parse the downloaded content and extract structured data.
+- When data is extracted from an unstructured form into a structured form, the scraper saves it in one of several formats, such as CSV, JSON, or a database.
```
Comment (Bayler) [inline content/articles/understanding-gazpacho-in-python-and-its-application-in-webscraping/index.md:54]:
```suggestion
- When we extract data from an unstructured into a structured form, the scraper saves it in one of several formats, such as CSV, JSON, or a database.
```

File: content/articles/understanding-gazpacho-in-python-and-its-application-in-webscraping/index.md [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+- Extractor: This component processes the fetched HTML content from the web crawler module and extracts the data into a semi-structured format. It's also known as a parser module because it works with many parsing approaches, such as regular expressions, HTML parsing, DOM parsing, and Artificial Intelligence.
+- Data transformation and cleaning module: The data collected by the extractor isn't ready to use right away; it needs to go through some sort of cleaning process before we can utilize it. For this, methods such as string manipulation and regular expressions can be employed. Extraction and transformation can be done in one process.
+- Once we have extracted the needed data, it is stored in standard formats that can be stored in the database, in JSON, or CSV format in the storage module.
+- 
+#### Operation of the Web Scraper
+For the web scraper to perform and accomplish the scraping process, it goes through various steps as stated below:
+- First, the web scraper downloads the requested content from multiple websites and web pages through the use of a web crawler.
```
Comment (Bayler) [inline content/articles/understanding-gazpacho-in-python-and-its-application-in-webscraping/index.md:52]:
```suggested
- First, the web scraper downloads the requested content from multiple websites and web pages through a web crawler.
```

File: content/articles/understanding-gazpacho-in-python-and-its-application-in-webscraping/index.md [hunk 6]
```diff
@@ -0,0 +1,170 @@
...
+### Prerequisites
+To follow along with this tutorial, the reader should:
+- Have a prior understanding of the basic Python concepts.
+- Ensure Python is installed on the machines.
+- Install Jupyter Notebook or another IDE/editor that can run Python.
+
+Go to this [page](https://jupyter.org/install) to install Jupyter on your PC.
+
+### Web Scraping
+Web scraping, also known as web harvesting or web extraction, is commonly used when obtaining information from the internet. It is much quicker and easier than the manual methods since it is an automatic method of extracting data from a web page by querying a web server, requesting data (usually in the form of HTML and other files that comprise web pages), and then parsing it to extract the needed information. This is made possible using automated software called ***web scrapers***, which is dedicated to performing data harvesting.
```
Comment (Bayler) [inline content/articles/understanding-gazpacho-in-python-and-its-application-in-webscraping/index.md:41]: The sentence, "It is much quicker...." is too long. Consider breaking it into two parts.

**Comment by dentex22** [comment]
@Bayler done with the requested changes

**Comment by ahmadmardeni1** [comment]
Great work team. @dentex22 @Bayler

**Comment by hectorkambow** [comment]
@dentex22 please reach out to me via Slack DM asap 👍  

Thank you
